src/pages/Profile.tsx

```diff
@@ -159,18 +170,37 @@ const Profile = () => {
             <Box sx={styles.profileContent}>
               <Box sx={styles.avatarContainer}>
                 <Avatar sx={styles.avatar} />
-                {profileContents.username === user.username ? (
-                  <Button
-                    size="small"
-                    variant="outlined"
-                    startIcon={<EditIcon />}
-                    sx={styles.editProfileButton}
-                  >
-                    Edit Profile
-                  </Button>
-                ) : (
-                  <FollowButton />
-                )}
+                {!loading &&
+                  (profileContents.username === user.username ? (
+                    <Button
+                      size="small"
+                      variant="outlined"
+                      startIcon={<EditIcon />}
+                      sx={styles.editProfileButton}
+                    >
+                      Edit Profile
+                    </Button>
+                  ) : profileContents.followStatus ? (
+                    <FollowingButton
+                      onClick={() =>
+                        setProfileContents({
+                          ...profileContents,
+                          followStatus: false,
+                        })
+                      }
+                      username={username}
+                    />
+                  ) : (
+                    <FollowButton
+                      onClick={() =>
+                        setProfileContents({
+                          ...profileContents,
+                          followStatus: true,
+                        })
+                      }
+                      username={username}
+                    />
```

Comment:
One other thing, we forgot to update the follower count after clicking the follow/unfollow button

Reply:
updated! but its a bit different to how I'd normally do it as:
`followerCount: profileContents.followerCount + 1` 
does not increment the count but appends another 1 onto the number, so a 1 follower account would show 11 followers, or 2 would show 21. 

I decided to use pre-incrementing because of the nature of setState, it wasn't updating as it should

Reply:
We need to look into why `followerCount` changes type after the fetch. New ticket needed

Reply:
Yeah, we looked into this offline but couldn't find out why, so let's that works and explore this bug in another ticket